packages/graphql-hooks/test/unit/useClientRequest.test.js

```diff
@@ -205,6 +205,30 @@ describe('useClientRequest', () => {
       expect(state).toEqual({ cacheHit: false, loading: false, data: 'data' })
     })
 
+    it('calls request with options & doesnt update state if component unmounts whilst request is in flight', async () => {
+      let fetchData, state
+      const { unmount } = renderHook(
+        () =>
+          ([fetchData, state] = useClientRequest(TEST_QUERY, {
+            variables: { limit: 2 },
+            operationName: 'test'
+          })),
+        {
+          wrapper: Wrapper
+        }
+      )
+
+      const fetchDataPromise = fetchData()
+      unmount()
+      await fetchDataPromise
```

Comment:
In this test it should resolve the result, right? We can then add a test that after unmount it resolves undefined:

```js
const fetchDataPromise = fetchData()
unmount()
const firstResult = await fetchDataPromise // <- expected actual result
const secondResult = await fetchData() // <- expected undefined
```

Reply:
I've added a test for that 👍